use-cases/client-server-msgs-perf/svmsg_file_server.c

```diff
@@ -543,22 +615,26 @@ svr_op_ct_init(requestMsg *req)
     resp.clientId = req->clientId;   // For this client
 
     // Use the next free slot w/o trying to recycle existing free slots.
-    resp.client_idx = NumActiveClientsHWM;
+    resp.client_idx = NumActiveClientsHWM++;
     resp.counter = req->counter;     // Just init'ed; no incr/decr done
 
     // Save off client's initial state
-    Client_info *clientp = &ActiveClients[NumActiveClientsHWM];
+    Client_info *clientp = &ActiveClients[resp.client_idx];
     memset(clientp, 0, sizeof(*clientp));
 
     clientp->clientId   = req->clientId;
-    clientp->client_idx = NumActiveClientsHWM;
+    clientp->client_idx = resp.client_idx;
     clientp->client_ctr = req->counter;
     clientp->last_mtype = req->mtype;
 
     NumActiveClients++;
-    NumActiveClientsHWM++;
```

Comment:
NOTE: There is a potential race on prev L630 where `NumActiveClientsHWM` is incremented earlier, but `NumActiveClients` is incremented above.

In theory, both should be incremented in lock-step.

In practice, it should not matter critically if the client's index and `NumActiveClients` don't quite sync-up.

I have not found anything untoward so far with this approach, so will let this be, to see how things fare in further testing.

Reply:
Agree, I don't see a problem with `NumActiveClientsHWM` and `NumActiveClients` being out of sync.